### PR TITLE
Minor updates

### PR DESCRIPTION
**Pull request by SergioGasquez**
- Fix format in a few files
- Update deps
- `verify.sh` now exits on errors

Supersedes #135

### CODE REVIEW

**Comment by bjoernQ** [comment]
I think it will cause problems with `esp-wifi` since `0.1.1` needs the older HAL. We could make it a git-dependency and patch a lot of the dependencies like in https://github.com/esp-rs/esp-wifi/blob/12477b3b5fd3b3b17727a0471b624eb5b0578f6e/Cargo.toml#L59-L65

Really hope our dependencies get a release and we are able to release `esp-wifi 0.2.0` since it's blocking a few things

**Comment by MabezDev** [comment]
Hopefully we will have a enet release soon: https://github.com/embassy-rs/embassy/pull/2401 then we can bump esp-wifi and merge this.

**Comment by MabezDev** [comment]
embassy-net 0.3 is now available on crates.io so I think we can get things moving here!

**Comment by bjoernQ** [comment]
> embassy-net 0.3 is now available on crates.io so I think we can get things moving here!

Awesome! I will update + release esp-wifi tomorrow

File: Cargo.toml [hunk 1]
```diff
@@ -6,21 +6,31 @@ edition = "2021"
 license = "MIT OR Apache-2.0"
 
 [dependencies]
-hal = { package = "{{ mcu }}-hal", version = "{{ hal_version }}" }
+{{ mcu }}-hal = "{{ hal_version }}"
```
Comment (bjoernQ) [inline Cargo.toml:9]: What's the reason we don't want to alias the HAL dependency anymore?

Comment (jessebraham) [reply Cargo.toml:9]: What was the reason we did it in the first place? I always found it weird and needless, every time I generated a template I changed it.

Comment (bjoernQ) [reply Cargo.toml:9]: Oh ... interesting. I found it to be easier to copy code around this way ... however, I don't have a hard preference. Was just curious - and also once we did the split, we will just have `esp_hal` or something

Comment (jessebraham) [reply Cargo.toml:9]: I don't feel too strongly about this, so feel free to revert if anybody else does.

Comment (SergioGasquez) [reply Cargo.toml:9]: I dont have any hard preference, tbh

File: Cargo.toml [hunk 2]
```diff
@@ -6,21 +6,26 @@ edition = "2021"
...
-esp-wifi  = { version = "0.1.1", features = ["{{ mcu }}", "wifi"] }
-smoltcp = { version = "0.10.0", default-features=false, features = ["proto-igmp", "proto-ipv4", "socket-tcp", "socket-icmp", "socket-udp", "medium-ethernet", "proto-dhcpv4", "socket-raw", "socket-dhcpv4"] }
-embedded-svc = { version = "0.26.1", default-features = false, features = [] }
-embedded-io = "0.4.0"
-heapless = { version = "0.7.14", default-features = false }
+esp-wifi  = { version = "0.2.0", features = ["{{ mcu }}", "wifi"] }
+smoltcp = { version = "0.11.0", default-features=false, features = ["proto-igmp", "proto-ipv4", "socket-tcp", "socket-icmp", "socket-udp", "medium-ethernet", "proto-dhcpv4", "socket-raw", "socket-dhcpv4"] }
+embedded-svc = { version = "0.26.4", default-features = false, features = [] }
+embedded-io = "0.6.1"
+heapless = { version = "0.8.0", default-features = false }
```
Comment (bjoernQ) [inline Cargo.toml]: I'm not sure we should bump this since it's mostly used for `embedded-svc` which is still on `0.7+`

Comment (bjoernQ) [reply Cargo.toml]: oh sorry - we shouldn't downgrade embedded-io but heapless: https://crates.io/crates/embedded-svc/0.26.4/dependencies

Reason is, it's used in esp-wifi's (through embedded-svc) public API

Comment (bjoernQ) [reply Cargo.toml]: downgrading smoltcp is probably good since for some reason we forgot to bump it in esp-wifi 0.2.0 🤦‍♂️

Comment (SergioGasquez) [reply Cargo.toml]: I think now, if I understood it correctly, dependencies should be fine.

**Review by bjoernQ — APPROVED** [review]
LGTM - Thanks

File: Cargo.toml (hunk 1 above)
Comment (georgik) [reply Cargo.toml:9]: The main idea behind alias is to increase portability of the code. Using alias makes it easier to have common app code for multiple targets.

Comment (MabezDev) [reply Cargo.toml:9]: The next release will unify the hal, meaning all hals will be exposed under `esp_hal`, just like `esp_idf_hal` supports all target in one.

**Pull request merged**